draft-ietf-core-resource-directory.md

```diff
@@ -1703,6 +1703,51 @@ Care must be taken in such setups to determine the applicable access control mea
 One easy way to do that is to mandate the use of the sector parameter on all operations,
 as no credentials are suitable for operations across sector borders anyway.
 
+## First-Come-First-Remembered: A default policy
+
+The First-Come-First-Remembered policy is provided both as a reference example for a security policy definition,
+and as a policy that implementations may choose to use as default policy in absence of other configuration.
+It is designed to enable efficient discovery operations even in ad-hoc settings.
+
+Under this policy, the RD accepts registrations for any endpoint name that is not assigned to an active registration resource,
+and only accepts registration updates from the same endpoint.
+The policy is minimal in that towards lookup clients it does not make any of the claims of {{entered-resources}} and {{link-confidentiality}},
+and its claims on {{secure-ep}} are limited to the lifetime of that endpoint's registration.
+It does, however, guarantee towards any endpoint that for the duration of its registration, its links will be discoverable on the RD.
+
+When a registration or operation is attempted, the RD MUST determine the client's subject name or public key:
+
+* If the client's credentials indicate a subject name certified by an authority the RD recognizes (which may be the system's trust anchor store), that subject name is stored.
+  With COSE based credentials (like in ACE), the name is taken from the Subject (sub) claim; this step fails if no Subject claim is part of the token.
+  With X.509 certificates, the Common Name (CN) and the complete list of SubjectAltName entries are stored.
+  In both cases, the authority that certified the claim is stored along the subject, as the latter may only be locally unique.
+* Otherwise, if the client's credentials contain a public key, that public key is stored.
+  With (D)TLS, this applies both to raw public keys and to the public keys indicated in certificates that are not recognized by the RD.
+* If neither is present, a reference to the security session itself is stored.
+  With (D)TLS, that is the connection itself, or the session resumption information if available.
+  With OSCORE, that is the security context.
+
+As part of the registration operation, that information is stored along the registration resource.
+
+The RD MUST accept all registrations that do not conflict with an already active registration resource,
+as long as they are made using a security layer supported by the RD.
+
+Any operation on a registration resource,
+including registrations that lead to an existing registration resource,
+MUST be rejected by the RD unless the credentials information is identical to (or, in case of SubjectAltNames, a superset of) the stored information.
```

Comment:
The whole credentials information or the stored name?
(I'm not sure that "the" credentials information is defined here, either.)

Reply:
Mh, I had the extraction steps phrased as "MUST determine ...", and considered changing that to "MUST extract" so I can refer to "the extracted information" here -- but that's not quite it: If the client originally just had a certificate without a good trust chain (so the public key was stored), and later comes with a good certificate, the key should still be enough, so it became:

"unless all the stored information is found in the new request's credentials."